packages/sdk-react/src/components/receivables/InvoicesTable/useInvoiceRowActionMenuCell.tsx

```diff
@@ -0,0 +1,252 @@
+    cancel: isAllowedInvoiceAction(ActionEnum.UPDATE),
+    markUncollectible: isAllowedInvoiceAction(ActionEnum.UPDATE),
+    recurrent: isAllowedInvoiceAction(ActionEnum.CREATE),
+    partiallyPay: isAllowedInvoiceAction(ActionEnum.UPDATE),
+    pay: isAllowedInvoiceAction(ActionEnum.UPDATE),
+    overduePayment: isAllowedInvoiceAction(ActionEnum.UPDATE),
+    send: isAllowedInvoiceAction(ActionEnum.CREATE),
+  };
+
+  return menuItemsToFilter.filter(
+    (itemValue) => menuItemsPermissionMap[itemValue]
+  );
+};
+
+const DEFAULT_ACTION_LIST: InvoicesTableRowActionSchema = {
+  [ReceivablesStatusEnum.DRAFT]: ['view', 'edit', 'delete'], // 'issue', 'recurrent' are not default
+  [ReceivablesStatusEnum.ISSUED]: ['view', 'send', 'cancel'], // 'copyPaymentLink', 'partiallyPay', 'overduePayment' are not default
+  [ReceivablesStatusEnum.CANCELED]: ['view'],
+  [ReceivablesStatusEnum.PARTIALLY_PAID]: [
+    // 'copyPaymentLink', 'pay', 'overduePayment' are not default
+    'view',
+    'send',
+    'cancel',
+  ],
+  [ReceivablesStatusEnum.OVERDUE]: ['view', 'send', 'cancel'], // 'copyPaymentLink', 'pay', 'markUncollectible' are not default
+  [ReceivablesStatusEnum.PAID]: ['view'],
+  [ReceivablesStatusEnum.UNCOLLECTIBLE]: ['view'],
+  [ReceivablesStatusEnum.EXPIRED]: ['view'],
+  [ReceivablesStatusEnum.ACCEPTED]: ['view'],
+  [ReceivablesStatusEnum.DECLINED]: ['view'],
+  [ReceivablesStatusEnum.RECURRING]: ['view'],
+  [ReceivablesStatusEnum.DELETED]: [],
+};
+
+const getInvoiceActionMenuItemLabels = (
+  i18n: I18n
+): Record<InvoicesTableRowAction, string> => ({
+  view: t(i18n)({
+    message: 'View',
+    context: 'InvoicesTableRowActionMenu',
+  }),
+  send: t(i18n)({
+    message: 'Send',
+    context: 'InvoicesTableRowActionMenu',
+  }),
+  copyPaymentLink: t(i18n)({
+    message: 'Copy payment link',
+    context: 'InvoicesTableRowActionMenu',
+  }),
+  cancel: t(i18n)({
+    message: 'Cancel with credit note',
+    context: 'InvoicesTableRowActionMenu',
+  }),
+  edit: t(i18n)({
+    message: 'Edit',
+    context: 'InvoicesTableRowActionMenu',
+  }),
+  delete: t(i18n)({
+    message: 'Delete',
+    context: 'InvoicesTableRowActionMenu',
+  }),
+  markUncollectible: t(i18n)({
+    message: 'Mark uncollectible',
+    context: 'InvoicesTableRowActionMenu',
+  }),
+  issue: t(i18n)({
+    message: 'Issue',
+    context: 'InvoicesTableRowActionMenu',
+  }),
+  recurrent: t(i18n)({
+    message: 'Recurring',
+    context: 'InvoicesTableRowActionMenu',
+  }),
+  partiallyPay: t(i18n)({
+    message: 'Partially pay',
+    context: 'InvoicesTableRowActionMenu',
+  }),
+  pay: t(i18n)({
+    message: 'Pay',
+    context: 'InvoicesTableRowActionMenu',
+  }),
+  overduePayment: t(i18n)({
+    message: 'Overdue payment',
+    context: 'InvoicesTableRowActionMenu',
+  }),
+});
+
+interface InvoicesTableRowActionMenuItem {
+  label: string;
+  action: InvoicesTableRowAction;
+}
+
+export type InvoicesTableRowAction =
```

Comment:
It would be nice to provide TSDoc for each of options. As I see this type will be provided to the customer and I think it's better to provide TSDoc for each of actions (especially custom ones) to get the customer know what & how it will affect them

Reply:
Added, thanx!